tests/jointcalTestBase.py

```diff
@@ -0,0 +1,311 @@
+    # for i, ref in enumerate(dataRefs):
+    #     md = ref.get('calexp_md')
+    #     plot_wcs(plt, old_wcss[i], ref.get('wcs').getWcs(),
+    #              dim=(md.get('NAXIS1'), md.get('NAXIS1')),
+    #              center=(md.get('CRVAL1'), md.get('CRVAL2')), name='dataRef %d'%i)
+
+
+def wcs_convert(xv, yv, wcs):
+    """Convert two arrays of x/y points into an on-sky grid."""
+    xout = np.zeros((xv.shape[0], yv.shape[0]))
+    yout = np.zeros((xv.shape[0], yv.shape[0]))
+    for i, x in enumerate(xv):
+        for j, y in enumerate(yv):
+            sky = wcs.pixelToSky(x, y).toFk5()
+            xout[i, j] = sky.getRa()
+            yout[i, j] = sky.getDec()
+    return xout, yout
+
+
+def plot_all_wcs_quivers(plt, dataRefs, visitCatalogs, old_wcss):
+    """Make quiver plots of the WCS deltas for each CCD in each visit."""
+
+    for cat in visitCatalogs:
+        fig = plt.figure()
+        fig.set_tight_layout(True)
+        ax = fig.add_subplot(111)
+        for old_wcs, ref in zip(old_wcss, dataRefs):
+            if ref.dataId['visit'] == cat:
+                md = ref.get('calexp_md')
+                Q = plot_wcs_quivers(ax, old_wcs, ref.get('wcs').getWcs(),
+                                     dim=(md.get('NAXIS1'), md.get('NAXIS2')))
+                # TODO: add CCD bounding boxes to plot once DM-5503 is finished.
+                # TODO: add a circle for the full focal plane.
+        ax.quiverkey(Q, 0.9, 0.95, 0.1/arcsec_per_radian, '0.1 arcsec', coordinates='figure')
+        plt.xlabel('RA')
+        plt.ylabel('Dec')
+        plt.title('visit: {}'.format(cat))
+
+
+def plot_wcs_quivers(ax, wcs1, wcs2, dim=(1000, 1000)):
+    """Plot the delta between wcs1 and wcs2 as vector arrows."""
+
+    x = np.linspace(0, dim[0], 50)
+    y = np.linspace(0, dim[0], 50)
+    x1, y1 = wcs_convert(x, y, wcs1)
+    x2, y2 = wcs_convert(x, y, wcs2)
+    u = x2-x1
+    v = y2-y1
+    Q = ax.quiver(x1, y1, u, v, units='x', pivot='tail', scale=1e-3, width=5e-6)
+    return Q
+
+
+def plot_wcs(plt, wcs1, wcs2, dim=(1000, 1000), center=(0, 0), name=""):
+    """Plot the "distortion map": wcs1-wcs2 delta of points in the CCD grid."""
+
+    plt.figure()
+
+    x = np.linspace(0, dim[0], 50)
+    y = np.linspace(0, dim[0], 50)
+    x1, y1 = wcs_convert(x, y, wcs1)
+    x2, y2 = wcs_convert(x, y, wcs2)
+
+    plt.plot((x1-x2)*arcsec_per_radian + center[0], (y1-y2)*arcsec_per_radian + center[1], '-')
+    plt.xlabel('delta RA (arcsec)')
+    plt.ylabel('delta Dec (arcsec)')
+    plt.title(name)
+
+
+def plot_rms_histogram(plt, old_rms_relative, old_rms_absolute,
+                       new_rms_relative, new_rms_absolute,
+                       old_rel_total, old_abs_total, new_rel_total, new_abs_total):
+    """Plot histograms of the star separations and their RMS values."""
+    plt.ion()
+    plt.figure()
+
+    color_rel = 'black'
+    ls_old = 'dotted'
+    color_abs = 'blue'
+    ls_new = 'dashed'
+    lw = 2
```

Comment:
Suggest putting common options in a dict so you can use them via `**plotOptions`.